stores/wallet.ts

```diff
@@ -125,26 +125,31 @@ export const useWalletStore = defineStore('wallet', () => {
       }
       const session = connector.value.restoreSession()
       if (session) {
-        try {
-          const connection = await connector.value.initIfNecessary()
-          if (connection) { await initWallet(connection) }
-        } catch (e) {
-          console.error(e)
-        }
+        handleConnection({ accounts: session.accounts, method: session.method })
       }
     }
   }
 
-  function handleConnection (connection: LikeCoinWalletConnectorConnectionResult) {
-    if (!connection) { return }
-    accounts.value = connection.accounts
-    signer.value = connection.offlineSigner as (OfflineAminoSigner & OfflineDirectSigner)
-    const method = connection.method
-    useTrackEvent('login', { method })
+  function handleConnection (connection: LikeCoinWalletConnectorConnectionResult | { accounts: any[], offlineSigner?: OfflineAminoSigner & OfflineDirectSigner, method: string }) {
+    const accountsValue = connection.accounts
+    const signerValue = connection?.offlineSigner || undefined
+    const methodValue = connection.method
+
+    if (accountsValue.length === 0 || !methodValue) {
+      throw new Error("'accounts' or 'method' cannot be empty.")
+    }
```

Comment:
Not sure if we should just return instead of throw, since the old function does `if (!connection) { return }`